pkg/model/resource/options_test.go

```diff
@@ -73,24 +74,37 @@ var _ = Describe("Resource Options", func() {
 			Expect(options.Validate().Error()).To(ContainSubstring("kind cannot be empty"))
 		})
 
-		It("should fail if the Kind is not pascal cased", func() {
-			// Base case
-			options := &Options{Group: "crew", Version: "v1", Kind: "FirstMate"}
-			Expect(options.Validate()).To(Succeed())
-
-			// Can't detect this case :(
-			options = &Options{Group: "crew", Version: "v1", Kind: "Firstmate"}
-			Expect(options.Validate()).To(Succeed())
+		DescribeTable("valid Kind values - according to core Kubernetes",
+			func(kind string) {
+				options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+				Expect(options.Validate()).To(Succeed())
+			},
+			Entry("camelcase", "FirstMate"),
+			Entry("more than one caps at the start", "FIRSTMate"),
+		)
+
+		It("should fail if Kind is too long", func() {
+			var kind string
+
+			for i := 0; i < 65; i++ {
+				kind += "a"
+			}
+
+			options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+			err := options.Validate()
+			Expect(err).To(MatchError(ContainSubstring("must be no more than 63 characters")))
+		})
 
-			options = &Options{Group: "crew", Version: "v1", Kind: "firstMate"}
-			Expect(options.Validate()).NotTo(Succeed())
-			Expect(options.Validate().Error()).To(ContainSubstring(
-				`kind must be PascalCase (expected FirstMate was firstMate)`))
+		It("should fail if Kind is invalid, according to core Kubernetes", func() {
+			options := &Options{Group: "crew", Kind: "Something withSpaces", Version: "v1"}
+			err := options.Validate()
+			Expect(err).To(MatchError(ContainSubstring("a DNS-1035 label must consist of lower case alphanumeric characters")))
```

Comment:
I think you should add `Expect(err).To(HaveOccurred())` above this line to avoid panics in case the error is nil.

Reply:
`MatchError(...)` can handle `nil` errors gracefully from what I remember, so `HaveOccured()` is likely redundant 🤔 I'll double check

Reply:
Yep indeed (for cross reference: [gomega implementation](https://github.com/onsi/gomega/blob/6be6c439588487cd908a3a700795660c2a16dfec/matchers/match_error_matcher.go#L16-L18))

No super strong opinions though, happy to add it if there's significant value in adding the error expectation.

Reply:
Didn't know about this helper until today so nice to learn something new (I should probably apply it in the `pkg/scaffold/internal/machienry` package too). Probably the error messages would be a bit different but they should both clear enough.